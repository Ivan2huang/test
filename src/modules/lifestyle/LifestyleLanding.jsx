import React from 'react';
import * as PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';

import { Box } from '@material-ui/core';

import Images from '../../constants/images';
import Grid from '../../uiComponents/Grid';
import GridItem from '../../uiComponents/GridItem';
import Typography from '../../uiComponents/Typography';
import withLayout from '../../layouts/withLayoutProvider';
import LifestyleButtons from './LifestyleButtons';
import { formatMessage } from '../../helpers/helpers';

const LifestyleLanding = ({ intl }) => {
  return (
    <Box pt={{ xs: 10, md: 40 }}>
      <Grid>
        <GridItem columns={{ md: 4 }}>
          <img src={Images.LIFESTYLE_LANDING} alt="lifestyle landing" />
        </GridItem>
        <GridItem columns={{ xs: 12, md: 6 }} offset={{ md: 2 }}>
          <Typography type="style-1">
            {formatMessage(
              intl,
              'lifestyle.landingPage.header',
              'Become a better version of yourself !',
            )}
          </Typography>
          <Box mt={6}>
            <Typography type="style-6">
              {formatMessage(
                intl,
                'lifestyle.landingPage.description-first',
                'Discover actionable insights on your health status. Enjoy\n' +
                  'personalised recommendations to improve your daily wellbeing.',
              )}
            </Typography>
            <Box mt={4}>
              <Typography type="style-6">
                {formatMessage(
                  intl,
                  'lifestyle.landingPage.description-second',
                  'Donâ€™t worry, no data is shared with your employer.',
                )}
              </Typography>
            </Box>
          </Box>
          <Box mt={{ xs: 4, md: 10 }}>
            <LifestyleButtons />
          </Box>
        </GridItem>
      </Grid>
    </Box>
  );
};

LifestyleLanding.propTypes = {
  intl: PropTypes.shape({}).isRequired,
};

export default withLayout(injectIntl(LifestyleLanding));
