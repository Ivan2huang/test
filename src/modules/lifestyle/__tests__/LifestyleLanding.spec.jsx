/* eslint-disable react/prop-types */
import React from 'react';

import { render } from '@testing-library/react';

import withTheme from '../../../themes/withThemeProvider';
import LifestyleLanding from '../LifestyleLanding';
import withIntl from '../../../i18n/withIntlProvider';

jest.mock('../../../helpers/helpers', () => ({
  formatMessage: jest.fn((intl, id, defaultMessage) => defaultMessage),
}));

jest.mock('../../../uiComponents/Typography', () => ({ children, ...rest }) => (
  <div {...rest}>
    {children}
    (Typography)
  </div>
));

jest.mock('../../../uiComponents/GridItem', () => ({ columns, children }) => (
  <div data-xs={columns.xs} data-sm={columns.sm} data-md={columns.md}>
    <span>GridItem Component</span>
    <div>{children}</div>
  </div>
));

jest.mock('../../../uiComponents/Grid', () => ({ children }) => (
  <div>
    <span>Grid Component</span>
    <div>{children}</div>
  </div>
));

jest.mock('../LifestyleButtons', () => () => (
  <div>
    <span>add button</span>
    <span>clinic button</span>
  </div>
));

jest.mock('../../../layouts/withLayoutProvider', () => Component => () => (
  <div>
    <header>Layout Header</header>
    <Component />
  </div>
));

describe('No Lifestyle Component', () => {
  let result;

  beforeEach(() => {
    const Component = withIntl(withTheme(LifestyleLanding));
    result = render(<Component />);
  });

  it('should match snapshot', () => {
    const { container } = result;
    expect(container).toMatchSnapshot();
  });
});
