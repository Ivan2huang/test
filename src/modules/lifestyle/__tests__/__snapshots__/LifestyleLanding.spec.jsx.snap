// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`No Lifestyle Component should match snapshot 1`] = `
<div>
  <div>
    <header>
      Layout Header
    </header>
    <div
      class="MuiBox-root MuiBox-root-2"
    >
      <div>
        <span>
          Grid Component
        </span>
        <div>
          <div
            data-md="4"
          >
            <span>
              GridItem Component
            </span>
            <div>
              <img
                alt="lifestyle landing"
                src="/static/lifestyle_landing.svg"
              />
            </div>
          </div>
          <div
            data-md="6"
            data-xs="12"
          >
            <span>
              GridItem Component
            </span>
            <div>
              <div
                type="style-1"
              >
                Become a better version of yourself !
                (Typography)
              </div>
              <div
                class="MuiBox-root MuiBox-root-3"
              >
                <div
                  type="style-6"
                >
                  Discover actionable insights on your health status. Enjoy
personalised recommendations to improve your daily wellbeing.
                  (Typography)
                </div>
                <div
                  class="MuiBox-root MuiBox-root-4"
                >
                  <div
                    type="style-6"
                  >
                    Donâ€™t worry, no data is shared with your employer.
                    (Typography)
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root MuiBox-root-5"
              >
                <div>
                  <span>
                    add button
                  </span>
                  <span>
                    clinic button
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
